doc/en/example/markers.rst

```diff
@@ -200,6 +200,13 @@ Or to select "http" and "quick" tests:
 
 You can use ``and``, ``or``, ``not`` and parentheses.
 
+.. note::
+    ``-k`` option will match not only function names but also `function attributes <https://www.python.org/dev/peps/pep-0232/>`_ .
+    For example, every function has attributes such as ``__call__`` by default,
+    and some libraries like `hypothesis <https://github.com/HypothesisWorks/hypothesis>`_ attach additional attributes to functions.
+    So if you specify such a keyword, let's say ``-k "__call__"``, then ``-k`` will match all functions in the scope, because all functions have ``__call__`` attributes.
+    Unfortunetelly you can't controll this behavior in current version, so please keep it in your mind.
```

Comment:
I think we should definitely lay out what is included in `-k`. Below is a suggestion with these differences:

- Added some more things which are matched against.
- Removed the reference to the ancient PEP, I don't think it adds too much.
- Removed the `__call__` example. Normally `__call__` is not found in a function's `__dict__` so is not matched. Maybe Hypothesis does something more customized, but I think it will be confusing for other users.
- Removed the "unfortunately" sentence. At least in pytest the behavior is more of a feature than a bug. I think just explaining the behavior is enough.

```suggestion
    In addition to the test's name, ``-k`` also matches the names of the test's parents (usually, the name of the file and class it's in), attributes set on the test function, markers applied to it or its parents and any :attr:`extra keywords <_pytest.nodes.Node.extra_keyword_matches>` explicitly added to it or its parents.
```